internal/shared/validations/validations.go

```diff
@@ -0,0 +1,34 @@
+package validations
+
+import (
+	"regexp"
+
+	gogit "github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+)
+
+// CheckGitURL checks whether the given git url is a valid git url or not by fetching remote references
+// TODO: for private repository
+func CheckGitURL(gitURL string) bool {
+	rem := gogit.NewRemote(nil, &config.RemoteConfig{
+		Name: "origin",
+		URLs: []string{gitURL},
+	})
```

Comment:
hmm, I wonder if we want to do this....from the validation I've seen before, it's always fairly simple and there's no reaching out across the network. We also wouldn't want the case where some intermittent network error causes us to fail validation, which I think could happen here

Reply:
Yeah i see ,so we should not look for the valid github repo url and wait till user gets the error while cloning ? @juliev0

Reply:
ok ,i will remove the network part ,but still there should be some check to validate whether the git url is proper ,something like a regex check . As git url can be with https or git:// ,I was checking out this stackoverflow question -https://stackoverflow.com/questions/31801271/what-are-the-supported-git-url-formats ,should we implement something like this @juliev0

Reply:
yes, that sounds good - thanks